app/scripts/metamask-controller.js

```diff
@@ -74,6 +76,21 @@ module.exports = class MetamaskController extends EventEmitter {
    */
   constructor (opts) {
     super()
+    chrome.runtime.onMessage.addListener((request, sender, sendResponse) => {
```

Comment:
We currently try to keep platform-specific API usage outside of the metamask-controller, so that it can be more platform-portable.

The way we currently do this is using the `platform` object passed into the MetaMask controller's constructor options.

In this case, I think an easier (and more flexible) approach would be putting this logic in `background.js`, which already has a `runtime.onConnectExternal.addListener` callback, which I think is a perfect place to add this conditional platform-distinguishing logic:
https://github.com/MetaMask/metamask-extension/blob/develop/app/scripts/background.js#L320